src/mesh/gmsh_io.C

```diff
@@ -275,10 +283,96 @@ void GmshIO::read_mesh(std::istream & in)
                 }
             }
 
+          else if (s.find("$Entities") == static_cast<std::string::size_type>(0))
+          {
+            if (version >= 4.0)
+            {
+              size_t num_point_entities, num_curve_entities, num_surface_entities, num_volume_entities;
+              in >> num_point_entities >> num_curve_entities >> num_surface_entities >> num_volume_entities;
+
+              for (size_t n = 0; n < num_point_entities; ++n)
+              {
+                int point_tag, physical_tag;
+                Real x, y, z;
+                size_t num_physical_tags;
+                in >> point_tag >> x >> y >> z >> num_physical_tags;
+                if (num_physical_tags > 1)
+                  libmesh_error_msg("I don't believe that we can specify multiple subdomain or " <<
```

Comment:
We definitely can't (currently) do this for the Elem::subdomain_id, but if we find ourselves erroring out a lot of files we might be able to handle this by storing the additional IDs in some other data structure that we could make available to the user.

Anyway I'd change the language of these error messages from "I don't believe we can" to simply "Cannot"!